OnTheMapTest/OTMLoginViewController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  ViewController.swift
+//  OnTheMapTest
+//
+//  Created by Gareth O'Sullivan on 17/12/2018.
+//  Copyright © 2018 Locust Redemption. All rights reserved.
+//
+
+import UIKit
+
+class OTMLoginViewController: UIViewController {
+    
+    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
+    @IBAction func Login() {
+        var request = URLRequest(url: URL(string: "https://onthemap-api.udacity.com/v1/session")!)
+        request.httpMethod = "POST"
+        request.addValue("application/json", forHTTPHeaderField: "Accept")
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.httpBody = "{\"udacity\": {\"username\": \"\(emailTextField.text!)\", \"password\": \"\(passwordTextField.text!)\"}}".data(using: .utf8)
```

Comment:
I'd still recommend using a dictionary which you convert to JSON. But it's up to you!

```
var body  = ["udacity": [
"username": emailTextField.text!,
"password": passwordTextField.text!
]]
do {
   request.httpBody = try JSONSerialization.data(withJSONObject: body, options: .prettyPrinted)
} catch {
    print(error.localizedDescription)
}
```

This is all 'guess code' as I don;t write in swift!

Reply:
Ah, I see what you mean now. I will try to implement this (and test your guesswork!)